server/server.go

```diff
@@ -0,0 +1,65 @@
+package server
+
+import (
+	"net/http"
+	"github.com/gruntwork-io/health-checker/options"
+	"net"
+	"fmt"
+)
+
+func StartHttpServer(opts *options.Options) {
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		checkTcpPorts(w, r, opts)
+	})
+	http.ListenAndServe(opts.Listener, nil)
+}
+
+// Check that we can open a TPC connection to all the ports in opts.Ports
+func checkTcpPorts(w http.ResponseWriter, r *http.Request, opts *options.Options) {
+	logger := opts.Logger
+	logger.Infof("Received inbound request. Beginning health checks...")
+
+	allPortsValid := true
+
+	for _, port := range opts.Ports {
+		err := attemptTcpConnection(port, opts)
```

Comment:
Should these checks happen in parallel? Imagine you have 5 ports and each one takes 1 second to respond... If your health check timeout was 3 seconds, this would fail, even though nothing was wrong.

Reply:
Good idea. Updated in 5f7842e9e1198f50771789c8fa8501d732e802bb.

Reply:
Note that I didn't put a Mutex on the `allPortsValid` variable, but the logic is such that the only value that would ever be written by the go routines is `false`.